src/sinol_make/commands/gen/__init__.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import glob
+import os
+import hashlib
+import yaml
+
+import multiprocessing as mp
+
+from sinol_make import util
+from sinol_make.commands.gen import gen_util
+from sinol_make.commands.gen.structs import OutputGenerationArguments
+from sinol_make.helpers import parsers, package_util, compile
+from sinol_make.interfaces.BaseCommand import BaseCommand
+
+
+class Command(BaseCommand):
+    """
+    Class for `gen` command.
+    """
+
+    def get_name(self):
+        return "gen"
+
+    def configure_subparser(self, subparser):
+        parser = subparser.add_parser(
+            self.get_name(),
+            help='Generate input and output files',
+            description='Generate input files using ingen program '
+                        '(for example prog/abcingen.cpp for abc task). Whenever '
+                        'the new input differs from the previous one, '
+                        'the model solution will be used to generate the new output '
+                        'file. You can also specify your ingen source '
+                        'file which will be used.'
+        )
+
+        parser.add_argument('ingen_path', type=str, nargs='?',
+                            help='path to ingen source file, for example prog/abcingen.cpp')
+        parser.add_argument('-c', '--cpus', type=int,
+                            help=f'number of cpus to use, by default {mp.cpu_count()} (all available). '
+                                 f'Used when generating output files.', default=mp.cpu_count())
+        parsers.add_compilation_arguments(parser)
+
+    def generate_outputs(self, outputs_to_generate):
+        print(f'Generating output files for {len(outputs_to_generate)} tests on {self.args.cpus} cpus.')
+        arguments = []
+        for output in outputs_to_generate:
+            output_basename = os.path.basename(output)
+            input = os.path.join(os.getcwd(), 'in', os.path.splitext(output_basename)[0] + '.in')
+            arguments.append(OutputGenerationArguments(self.correct_solution_exe, input, output))
+
+        results = []
+        with mp.Pool(self.args.cpus) as pool:
+            for i, result in enumerate(pool.imap(gen_util.generate_output, arguments)):
+                results.append(result)
+                if result:
+                    print(util.info(f'Successfully generated output file {os.path.basename(arguments[i].output_test)}'))
+                else:
+                    print(util.error(f'Failed to generate output file {os.path.basename(arguments[i].output_test)}'))
+
+        if not all(results):
+            util.exit_with_error('Failed to generate some output files.')
+        else:
+            print(util.info('Successfully generated all output files.'))
```

Comment:
```suggestion
        with mp.Pool(self.args.cpus) as pool:
            results = []
            for i, result in enumerate(pool.imap(gen_util.generate_output, arguments)):
                results.append(result)
                if result:
                    print(util.info(f'Successfully generated output file {os.path.basename(arguments[i].output_test)}'))
                else:
                    print(util.error(f'Failed to generate output file {os.path.basename(arguments[i].output_test)}'))

            if not all(results):
                util.exit_with_error('Failed to generate some output files.')
            else:
                print(util.info('Successfully generated all output files.'))
```

Reply:
Reducing the scope of a variable greatly reduces the size of context needed to read the whole code.